maths/series/geometric_series.py

```diff
@@ -11,53 +10,54 @@
 """
 
 
-def geometric_series(nth_term: int, start_term_a: int, common_ratio_r: int) -> list:
-    """Pure Python implementation of Geometric Series algorithm
+def geometric_series(
+    nth_term: float, start_term_a: float, common_ratio_r: float
+) -> list[float]:
+    """
+    Pure Python implementation of Geometric Series algorithm
+
     :param nth_term: The last term (nth term of Geometric Series)
     :param start_term_a : The first term of Geometric Series
     :param common_ratio_r : The common ratio between all the terms
     :return: The Geometric Series starting from first term a and multiple of common
         ration with first term with increase in power till last term (nth term)
+
     Examples:
-    >>> geometric_series(4, 2, 2)
-    [2, '4.0', '8.0', '16.0']
     >>> geometric_series(4.0, 2.0, 2.0)
-    [2.0, '4.0', '8.0', '16.0']
+    [2.0, 4.0, 8.0, 16.0]
+    >>> geometric_series(4.0, 2.0, 2.0)
+    [2.0, 4.0, 8.0, 16.0]
     >>> geometric_series(4.1, 2.1, 2.1)
-    [2.1, '4.41', '9.261000000000001', '19.448100000000004']
-    >>> geometric_series(4, 2, -2)
-    [2, '-4.0', '8.0', '-16.0']
-    >>> geometric_series(4, -2, 2)
-    [-2, '-4.0', '-8.0', '-16.0']
-    >>> geometric_series(-4, 2, 2)
+    [2.1, 4.41, 9.261000000000001, 19.448100000000004]
+    >>> geometric_series(4.0, 2.0, -2.0)
+    [2.0, -4.0, 8.0, -16.0]
+    >>> geometric_series(4.0, -2.0, 2.0)
+    [-2.0, -4.0, -8.0, -16.0]
+    >>> geometric_series(-4.0, 2.0, 2.0)
     []
-    >>> geometric_series(0, 100, 500)
+    >>> geometric_series(0, 100.0, 500.0)
     []
-    >>> geometric_series(1, 1, 1)
-    [1]
+    >>> geometric_series(1.0, 1.0, 1.0)
+    [1.0]
     >>> geometric_series(0, 0, 0)
     []
     """
-    if "" in (nth_term, start_term_a, common_ratio_r):
-        return ""
-    series = []
-    power = 1
+    if 0 in (nth_term, start_term_a, common_ratio_r):
```

Comment:
Let's test for all falsey values that the caller may provide.
```suggestion
    if not all((nth_term, start_term_a, common_ratio_r)):
```